test/cluster/mzcompose.py

```diff
@@ -2806,3 +2806,35 @@ def workflow_test_index_source_stuck(
         c.up("clusterd1")
         c.up("clusterd2")
         c.run("testdrive", "index-source-stuck/run.td")
+
+
+def workflow_test_github_cloud_7998(
+    c: Composition, parser: WorkflowArgumentParser
+) -> None:
+    """Regression test for MaterializeInc/cloud#7998."""
+
+    c.down(destroy_volumes=True)
+
+    with c.override(
+        Testdrive(no_reset=True),
+        Clusterd(name="clusterd1"),
+        Materialized(),
+    ):
+        c.up("materialized")
+        c.up("clusterd1")
+
+        c.run("testdrive", "github-cloud-7998/setup.td")
+
+        # Make the compute cluster unavailable.
+        c.kill("clusterd1")
+        c.run("testdrive", "github-cloud-7998/check.td")
+
+        # Trigger an environment bootstrap.
```

Comment:
Is this test deterministic or is there a chance that will occasionally pass as a false positive even without your fix?

Reply:
I think it becomes deterministic because we restart environmentd twice. This all depends on frontiers advancing while others are stuck, so maybe there is a chance that everything just happens so fast that frontiers never advance. But I doubt it.